src/loaders/asylum_load.c

```diff
@@ -124,26 +137,30 @@ static int asylum_load(struct module_data *m, HIO_HANDLE *f, const int start)
 	D_(D_INFO "Stored patterns: %d", mod->pat);
 
 	for (i = 0; i < mod->pat; i++) {
+		uint8 *pos;
 		if (libxmp_alloc_pattern_tracks(mod, i, 64) < 0)
 			return -1;
 
-		for (j = 0; j < 64 * mod->chn; j++) {
+		if (hio_read(buf, 1, 2048, f) < 2048) {
+			D_(D_CRIT "read error at pattern %d", i);
+			return -1;
+		}
+
+		pos = buf;
+		for (j = 0; j < 64 * 8; j++) {
```

Comment:
is `8` guaranteed here?

Reply:
in any case, we may need similar changes in libmikmod.

Reply:
Yes, this format has a fixed 8 channels.

Reply:
ok.